src/ripple/net/impl/RPCCall.cpp

```diff
@@ -624,10 +624,49 @@ class RPCParser
         return parseAccountRaw1 (jvParams);
     }
 
-    // account_lines <account> <account>|"" [<ledger>]
+    // account_lines <account> <account>|"" [<ledger> [<limit>] [<marker>]]
     Json::Value parseAccountLines (Json::Value const& jvParams)
     {
-        return parseAccountRaw2 (jvParams, jss::peer);
+        std::array<char const* const, 2> accFields{{jss::account, jss::peer}};
+        auto const nParams = jvParams.size ();
+        Json::Value jvRequest (Json::objectValue);
+        for (auto i = 0; i < nParams; ++i)
+        {
+            std::string strParam = jvParams[i].asString ();
+
+            if (i==1 && strParam.empty())
+                continue;
+
+            // Parameters 0 and 1 are accounts
+            if (i < 2)
+            {
+                if (parseBase58<PublicKey> (
+                        TokenType::TOKEN_ACCOUNT_PUBLIC, strParam) ||
+                    parseBase58<AccountID> (strParam) ||
+                    parseGenericSeed (strParam))
+                {
+                    jvRequest[accFields[i]] = std::move (strParam);
+                }
+                else
+                {
+                    return rpcError (rpcACT_MALFORMED);
+                }
+            }
+            else
+            {
+                if(i==2){
+                    if (!jvParseLedger (jvRequest, strParam))
+                        return rpcError (rpcLGR_IDX_MALFORMED);
+                }else{
+                    if(strParam.size()==64){
+                        jvRequest[jss::marker]=std::move (strParam);
+                    }else{
+                        jvRequest[jss::limit]=jvParams[i].asInt ();
+                    }
+                }
+            }
+        }
+        return jvRequest;
```

Comment:
As is, it could also accept more parameters than allowed. I agree with @seelabs, the code is clearer with a switch statement and his example avoids those issues. Optionally, you can also validate some of the parameters.
eg.
```
case 3:
    if (! jvParams[3].isIntegral())
        return rpcError(rpcINVALID_PARAMS);
    jvRequest[jss::limit] = jvParams[3].asInt();
    break;
case 4:
    if (! jvParams[4].isString())
        return rpcError(rpcINVALID_PARAMS);
    jvRequest[jss::marker] = jvParams[4].asString();
    break;
```